### DISCUSSION TITLE
Termux build

### DISCUSSION THREAD

**Discussion by aicynide** [userdocs/qbittorrent-nox-static#140] [General]
Can you build it with https://github.com/termux/termux-docker using Github Actions for termux?

**Comment by aicynide** [comment]
In case you are wondering libtorrent-rasterbar is already packaged for termux

**Comment by aicynide** [comment]
@userdocs here's how to build qbittorrent-nox in termux or [termux-docker](https://github.com/termux/termux-docker)


```
pkg in qt5-qttools qt5-qtsvg
git clone https://github.com/qbittorrent/qBittorrent.git
cd qBittorrent/
git checkout tags/release-4.3.7

mkdir build
cd build
cmake -D VERBOSE_CONFIGURE=ON -D STACKTRACE=OFF -DGUI=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=off -Dstatic_runtime=on -DCMAKE_INSTALL_PREFIX="${PREFIX}" ..
```

Research - https://github.com/termux/termux-packages/issues/3327#issuecomment-914167110

**Reply by userdocs** [reply]
Maybe, I might look into this at some point but right now I'm not really going to do that. I'd be open to someone showing it can be done.

So maybe one day or maybe not. 100% it's not happening in the near future.

**Reply by aicynide** [reply]
Pull request to add `qbittorrent-nox` to the termux-packages repo https://github.com/termux/termux-packages/pull/19991

Termux package repo **build script** (found in PR) - https://github.com/termux/termux-packages/pull/19991/files#diff-7703ec51d10bc01ee89d7da17ea9290f0c345365960614c5195d04fb544b397b

Github CI successfully built `qbittorrent-nox` https://github.com/termux/termux-packages/actions/runs/8878010622/job/24372772631?pr=19991

Binaries I got from the above github CI for termux aarch64,x86_64,x86 and armv7 - https://github.com/aicynide/flood/releases/download/v4.7.0-1/qbittorrent-nox-binaries-termux.zip
(Note: you need to install `libtorrent-rasterbar` & `qt5-qtbase` , `libicu`, `openssl` before running the binaries. install by running `pkg update && pkg install libicu libtorrent-rasterbar openssl qt5-qtbase`  in termux )
